dev/com.ibm.ws.jpa.tests.jpa_32_fat/test-applications/jakartadata/src/io/openliberty/jpa/data/tests/web/JakartaDataRecreateServlet.java

```diff
@@ -562,6 +562,40 @@ public void testOLGH28925() throws Exception {
 
         assertEquals(4, primes.size());
     }
+    @Test
+    // "Reference issue: https://github.com/OpenLiberty/open-liberty/issues/30093"
+    public void testOLGH30093() throws Exception {
+        deleteAllEntities(Prime.class); // Cleanup any left over entities
+
+        Prime two = Prime.of(2, "II", "two");
+        Prime three = Prime.of(3, "III", "three");
+        Prime five = Prime.of(5, "V", "five");
+        Prime seven = Prime.of(7, "VII", "seven");
+
+        List<Prime> primes;
+
+        tx.begin();
+        em.persist(two);
+        em.persist(three);
+        em.persist(five);
+        em.persist(seven);
+        tx.commit();
+
+        tx.begin();
+        try {
+            primes = em.createQuery(
+                                    "SELECT ID(THIS) FROM Prime WHERE ID(THIS) < ?1 ORDER BY ID(THIS) DESC",
+                                    Prime.class)
```

Comment:
```suggestion
            List<Long> ids = em.createQuery(
                                    "SELECT ID(THIS) FROM Prime WHERE ID(THIS) < ?1 ORDER BY ID(THIS) DESC",
                                    Long.class)
```

This test does not follow the Jakarta Persistence 3.2 specification, please refer to this section: 
https://jakarta.ee/specifications/persistence/3.2/jakarta-persistence-spec-3.2#identifier-and-version-functions

```txt
The ID and VERSION functions evaluate to the primary key or version, respectively, of their argument, which must be an identification variable assigned an entity abstract schema type or [...]
```

Therefore you need to assert that `SELECT ID(THIS) FROM ...` returns a list of IDs and not a list of entities.